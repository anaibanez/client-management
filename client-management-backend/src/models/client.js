const Address = require('./address');
const Company = require('./company');

class Client {

  constructor(
      { id = 0, name = '', username = '', email = '', phone = '',
        website = '', address = {}, company = {} }) {
    this.id = id;
    this.name = name;
    this.username = username;
    this.email = email;
    this.phone = phone;
    this.website = website;

    this.address = new Address(address);
    this.company = new Company(company);
  }

}

module.exports = Client;
