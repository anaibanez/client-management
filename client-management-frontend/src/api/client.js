import Api from '../utils/api';

const api = new Api();
const baseUri = '/v1/client';

export const getClients = () => api.get(`${baseUri}`);

export const getClient = (id) => api.get(`${baseUri}/${id}`);

export const updateClient = (body) => api.put(`${baseUri}/${body.id}`, { body });

export const createClient = (body) => api.post(`${baseUri}`, { body });

export const deleteClient = (id) => api.delete(`${baseUri}/${id}`);
