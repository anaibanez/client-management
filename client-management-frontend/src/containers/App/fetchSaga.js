import { call, put } from 'redux-saga/effects';

import { showError } from './actions';

export default function* fetchApiSaga(apiRequest, resource, content, action) {
  try {
    yield put({
      type: `FETCH_${resource}_PENDING`,
    });
    const response = yield call(apiRequest, content);
    yield put({
      type: `FETCH_${resource}_FULFILLED`,
    });
    if (action) {
      yield put(action(response));
    }
    return response;
  } catch (error) {
    yield put({
      type: `FETCH_${resource}_REJECTED`,
    });
    yield put(showError(error));
    return error;
  }
}
