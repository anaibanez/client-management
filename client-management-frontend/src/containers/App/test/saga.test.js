/**
 * Tests for HomePage sagas
 */

import { put } from 'redux-saga/effects';
import { getClient } from '../../../api/client';

import fetchSaga from '../fetchSaga';


/* eslint-disable redux-saga/yield-effects */
describe('fetchSaga', () => {
  let fetchSagaGenerator;
  beforeEach(() => {
    fetchSagaGenerator = fetchSaga(getClient, 'client');

    const putDescriptor = fetchSagaGenerator.next('client').value;
    expect(putDescriptor).toMatchSnapshot();

    const selectDescriptor = fetchSagaGenerator.next(getClient).value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the fulfilled if it requests the data successfully', () => {
    const response = {
      id: 10,
      name: 'Clementina DuBuque',
      username: 'Moriah.Stanton',
      email: 'Rey.Padberg@karina.biz',
      address: {
        street: 'Kattie Turnpike',
        suite: 'Suite 198',
        city: 'Lebsackbury',
        zipcode: '31428-2261',
        geo: {
          lat: '-38.2386',
          lng: '57.2232',
        },
      },
      phone: '024-648-3804',
      website: 'ambrose.net',
      company: {
        name: 'Hoeger LLC',
        catchPhrase: 'Centralized empowering task-force',
        bs: 'target end-to-end models',
      },
    };
    const putDescriptor = fetchSagaGenerator.next(response).value;
    expect(putDescriptor).toEqual(put({ type: 'FETCH_client_FULFILLED' }));
  });
});

