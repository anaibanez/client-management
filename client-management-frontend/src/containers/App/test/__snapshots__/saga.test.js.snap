// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchSaga should dispatch the fulfilled if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "FETCH_client_PENDING",
    },
    "channel": null,
  },
}
`;

exports[`fetchSaga should dispatch the fulfilled if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
